lib/board.rb

```diff
@@ -0,0 +1,67 @@
+  class Board
+    attr_accessor :cells, :token
+  
+  def initialize
+    @cells = [' ', ' ', ' ', ' ', ' ', ' ',' ', ' ',' ']
+  end
+  
+  def reset!
+    @cells = [' ', ' ', ' ', ' ', ' ', ' ',' ', ' ',' ']
+  end
+  
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]}  \n-----------\n #{cells[3]} | #{cells[4]} | #{cells[5]}  \n-----------\n #{cells[6]} | #{cells[7]} | #{cells[8]} \n\n\n"
+  end
+  
+  def position(input)
+    pos = input.to_i 
+    pos -= 1
+    @cells[pos]
+  end  
+  
+  def full?
+    if @cells.find { |i| i == ' '}
+      false
+    else
+      true
+    end 
+  end  
+  
+  def turn_count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```